source/adapters/level_zero/queue.cpp

```diff
@@ -1323,6 +1352,8 @@ ur_result_t urQueueReleaseInternal(ur_queue_handle_t Queue) {
           }
   }
 
+  Queue->clearEndTimeRecordings();
```

Comment:
it should probably be inside of synchronize() where we are sure all prior events on that queue completed.

Reply:
It is also done in `synchronize()`, but I don't see any guarantee that it is called before this function runs. Maybe I'm wrong?

Reply:
It's called in `urQueueRelease` - https://github.com/oneapi-src/unified-runtime/blob/main/source/adapters/level_zero/queue.cpp#L414.

I'm not sure whether it's possible for the queue to be unsynchronized in the urQueueReleaseInternal. urQueueRelease had to have been called already (to decrement the refcount).

Reply:
From what I can see, `urQueueReleaseInternal` is also called from within events: https://github.com/oneapi-src/unified-runtime/blob/main/source/adapters/level_zero/event.cpp#L888

They don't seem to call `synchronize` beforehand, for all paths.

Reply:
Ok, there's no harm in calling it twice to be sure.
I think for the queue to be released in the event, it had to have been released by the user already.